packages/jest-matchers/src/index.js

```diff
@@ -84,6 +106,65 @@ const getMessage = message => {
   return message;
 };
 
+const makeResolveMatcher = (
+  matcherName: string,
+  matcher: RawMatcherFn,
+  isNot: boolean,
+  actual: Promise<any>
+): PromiseMatcherFn => async (...args) => {
+  const matcherStatement = `.resolves.${isNot ? 'not.' : ''}${matcherName}`;
+  if (!isPromise(actual)) {
+    throw new JestAssertionError(
+      utils.matcherHint(matcherStatement, 'received', '') + '\n\n' +
+      `${utils.RECEIVED_COLOR('received')} value must be a Promise.\n` +
+      utils.printWithType('Received', actual, utils.printReceived),
+    );
+  }
+
+  let result;
+  try {
+    result = await actual;
+  } catch (e) {
+    throw new JestAssertionError(
+      utils.matcherHint(matcherStatement, 'received', '') + '\n\n' +
+      `Expected ${utils.RECEIVED_COLOR('received')} Promise to resolve, ` +
+      'instead it rejected to value\n' +
+      `  ${utils.printReceived(result)}`
+    );
+  }
+  return makeThrowingMatcher(matcher, isNot, result)(...args);
```

Comment:
I guess this is some bug with `transform-es2015-destructuring` and it doesn't transform some spread operators. You can use `.apply(args)` instead.